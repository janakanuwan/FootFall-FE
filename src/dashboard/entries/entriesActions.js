import { createAction } from 'redux-actions';
import { ADD_ENTRIES, CHANGE_ENTRIES_FROM, CHANGE_ENTRIES_TO } from '../../const/action-types';

const changeEntriesFrom = createAction(
  CHANGE_ENTRIES_FROM,
  time => time,
);

const changeEntriesTo = createAction(
  CHANGE_ENTRIES_TO,
  time => time,
);

const addEntries = createAction(
  ADD_ENTRIES,
  data => data,
);

export { changeEntriesFrom, changeEntriesTo, addEntries };
