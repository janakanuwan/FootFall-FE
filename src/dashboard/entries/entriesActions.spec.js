import { addEntries, changeEntriesFrom, changeEntriesTo } from './entriesActions';

describe('entriesActions', () => {

  it('should create an action to change entries from', () => {
    expect(changeEntriesFrom(+new Date('2018-04-24'))).toMatchSnapshot();
  });

  it('should create an action to change entries to', () => {
    expect(changeEntriesTo(+new Date('2018-04-28'))).toMatchSnapshot();
  });

  it('should create an action to add dashboard data', () => {
    expect(addEntries([{ id: 1, locationId: 1, entry: 1, exit: 2, time: 12345656 }]))
      .toMatchSnapshot();
  });

  it('should create an action to set failed dashboard data', () => {
    const action = addEntries(new Error('Failed to fetch'));
    expect(action).toMatchSnapshot();
    expect(action.error).toEqual(true);
    expect(action.payload.message).not.toEqual(null);
  });
});
