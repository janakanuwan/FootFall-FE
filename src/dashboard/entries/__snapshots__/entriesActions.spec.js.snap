// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entriesActions should create an action to add dashboard data 1`] = `
Object {
  "payload": Array [
    Object {
      "entry": 1,
      "exit": 2,
      "id": 1,
      "locationId": 1,
      "time": 12345656,
    },
  ],
  "type": "ADD_ENTRIES",
}
`;

exports[`entriesActions should create an action to change entries from 1`] = `
Object {
  "payload": 1524528000000,
  "type": "CHANGE_ENTRIES_FROM",
}
`;

exports[`entriesActions should create an action to change entries to 1`] = `
Object {
  "payload": 1524873600000,
  "type": "CHANGE_ENTRIES_TO",
}
`;

exports[`entriesActions should create an action to set failed dashboard data 1`] = `
Object {
  "error": true,
  "payload": [Error: Failed to fetch],
  "type": "ADD_ENTRIES",
}
`;
